main.py

```
@@ -5171,10 +5183,32 @@ async def modifica_parser_mio(slug: str, request: Request):
+        # niente TOCTOU — il controllo client-side puo' solo restringere la
+        # finestra, questo la chiude. `versione` avanza a OGNI scrittura,
+        # anche incondizionata: e' cio' che fa perdere in modo visibile la
+        # sessione rimasta indietro.
+        parametri = [titolo, json.dumps(dati.config), 1 if dati.active else 0,
+                     utente['id'], slug]
+        condizione = ''
+        if dati.versione is not None:
+            condizione = ' AND versione=?'
+            parametri.append(dati.versione)
+        esito = c.execute('UPDATE parsers SET titolo=?, config_json=?, active=?,'
+                          ' versione=versione+1 WHERE user_id=? AND slug=?' + condizione,
```

Ruff (0.16.1)

[error] Possible SQL injection vector through string-based query construction

(S608)